src/transactions/PathPaymentOpFrameBase.cpp

```diff
@@ -81,13 +81,17 @@ PathPaymentOpFrameBase::convert(AbstractLedgerTxn& ltx,
             if (offer.sellerID == getSourceID())
             {
                 // we are crossing our own offer
-                setResultOfferCrossSelf();
                 return OfferFilterResult::eStop;
             }
             return OfferFilterResult::eKeep;
         },
         offerTrail, maxOffersToCross);
 
+    if (r == ConvertResult::eFilterStop)
```

Comment:
should this just be moved to the switch below (line 103 in this commit)?

Reply:
I considered that but it's _technically_ a protocol change to do that. I'm comfortable with it if you are, so let me know.

Reply:
I don't understand why this would be a protocol change: the only difference in this function is that in case of exception below the result would not be set - but in that case, we end up with an internal error anyways.

Reply:
Oh yes, you're right. I'll make this change.